staging/src/kubesphere.io/utils/helm/executor.go

```diff
@@ -36,41 +57,488 @@ type Executor interface {
 	IsReleaseReady(timeout time.Duration) (bool, error)
 }
 
+const (
+	workspaceBaseSource = "/tmp/helm-executor-source"
+	workspaceBase       = "/tmp/helm-executor"
+
+	statusNotFoundFormat = "release: not found"
+	releaseExists        = "release exists"
+
+	kustomizationFile  = "kustomization.yaml"
+	postRenderExecFile = "helm-post-render.sh"
+	// kustomize cannot read stdio now, so we save helm stdout to file, then kustomize reads that file and build the resources
+	kustomizeBuild = `#!/bin/sh
+# save helm stdout to file, then kustomize read this file
+cat > ./.local-helm-output.yaml
+kustomize build
+`
+)
+
+var (
+	errorTimedOutToWaitResource = errors.New("timed out waiting for resources to be ready")
+)
+
 type executor struct {
+	// target cluster client
+	client.Client
+	KubeConfig string
+	Namespace  string
+	// helm release name
+	ReleaseName string
+	// helm action Config
+	helmConf  *action.Configuration
+	helmImage string
+	// add labels to helm chart
+	labels map[string]string
+	// add annotations to helm chart
+	annotations map[string]string
+	dryRun      bool
 }
 
 type Option func(*executor)
 
+// SetDryRun sets the dryRun option.
+func SetDryRun(dryRun bool) Option {
+	return func(wrapper *executor) {
+		wrapper.dryRun = dryRun
+	}
+}
+
+// SetAnnotations sets extra annotations added to all resources in chart.
+func SetAnnotations(annotations map[string]string) Option {
+	return func(wrapper *executor) {
+		wrapper.annotations = annotations
+	}
+}
+
+// SetLabels sets extra labels added to all resources in chart.
+func SetLabels(labels map[string]string) Option {
+	return func(wrapper *executor) {
+		wrapper.labels = labels
+	}
+}
+
+// SetHelmImage sets the helmImage option.
+func SetHelmImage(helmImage string) Option {
+	return func(wrapper *executor) {
+		wrapper.helmImage = helmImage
+	}
+}
+
 // NewExecutor generates a new Executor instance with the following parameters:
 //   - kubeConfig: kube config data of the target cluster
 //   - namespace: the namespace of the helm release
 //   - releaseName: the helm release name
 //   - options: functions to set optional parameters
 func NewExecutor(kubeConfig, namespace, releaseName string, options ...Option) (Executor, error) {
-	return &executor{}, nil
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig([]byte(kubeConfig))
+	if err != nil {
+		return nil, err
+	}
+	clusterClient, err := client.New(restConfig, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+
+	e := &executor{
+		KubeConfig:  kubeConfig,
+		Namespace:   namespace,
+		ReleaseName: releaseName,
+		Client:      clusterClient,
+		helmImage:   "iawia002/helm:test",
```

Comment:
We need to push this image to the `kubesphere` org.

Reply:
> We need to push this image to the `kubesphere` org.

`kubesphere/helm:latest` already pushed.

Reply:
Updated.